qcodes/instrument/visa.py

```diff
@@ -24,37 +24,36 @@ class VisaInstrument(Instrument):
     Base class for all instruments using visa connections.
 
     Args:
-        name (str): What this instrument is called locally.
-
-        address (str): The visa resource name to use to connect.
-            Optionally includes '@<backend>' at the end. For example,
-            'ASRL2' will open COM2 with the default NI backend, but
-            'ASRL2@py' will open COM2 using pyvisa-py. Note that qcodes
-            does not install (or even require) ANY backends, it is up to
-            the user to do that. see eg:
-            http://pyvisa.readthedocs.org/en/stable/names.html
-
-        timeout (int, float): seconds to allow for responses. Default 5.
-
+        name: What this instrument is called locally.
+        address: The visa resource name to use to connect.
+        timeout: seconds to allow for responses. Default 5.
         terminator: Read termination character(s) to look for. Default ``''``.
-
         device_clear: Perform a device clear. Default True.
-
+        visalib: Visa backend to use when connecting to this instrument.
+            This should be in the form of a string '@<backend>'.
```

Comment:
Why is this like that you may ask. I think we should remove this logic and just make it a string in a different pr

Reply:
do you mean the `@` logic? i believe that it comes from pyvisa

Reply:
Yes but that does not mean that it's not pointless :) As far as I can see the only thing `self.visalib` is used for is to split off the `@` to form `self.visabackend` in `set_adress`

Reply:
ok, understood. yes, i tend to agree. and then perhaps we can finally remove the logic about warnings and `@` being in `address` and `visalib` arguments.